### PR TITLE
Template change (minimum_version*) + `fmc_sgts`

### PR DESCRIPTION
**Pull request by rchrabas**
The following changes have been introduced:

### Template
Add support for the following template parameters:
- minimum_version_create
- minimum_version_bulk_create
- minimum_version_bulk_delete

Changed logic of bulk object generation: by default objects are bulk, unless overridden by minimum_version
Add handler for `ErrorCheck` in tests

### Types changed
Add support for `fmc_sgts` 
Updated definitions of objects with minimum_version*
Updated `fmc_network_groups` test definition

### CODE REVIEW

File: gen/templates/data_source.go [hunk 1]
```diff
@@ -184,6 +184,16 @@ func (d *{{camelCase .Name}}DataSource) Configure(_ context.Context, req datasou
 // Section below is generated&owned by "gen/generator.go". //template:begin read
 
 func (d *{{camelCase .Name}}DataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	{{- if .MinimumVersion}}
+	// Get FMC version
+	fmcVersion, _ := version.NewVersion(strings.Split(r.client.FMCVersion, " ")[0])
+
+	// Check if FMC client is connected to supports this object
+	if fmcVersion.LessThan(minFMCVersion{{camelCase .Name}}) {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("UnsupportedVersion: FMC version %s does not support {{.Name}}, minumum required version is {{.MinimumVersion}}", r.client.FMCVersion))
```
Comment (danischm) [inline gen/templates/data_source.go:193]: typo -> minumum

File: gen/templates/resource.go [hunk 2]
```diff
@@ -440,6 +446,16 @@ func (r *{{camelCase .Name}}Resource) Configure(_ context.Context, req resource.
 // Section below is generated&owned by "gen/generator.go". //template:begin create
 
 func (r *{{camelCase .Name}}Resource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	{{- if or .MinimumVersion .MinimumVersionCreate}}
+	// Get FMC version
+	fmcVersion, _ := version.NewVersion(strings.Split(r.client.FMCVersion, " ")[0])
+
+	// Check if FMC client is connected to supports this object
+	if fmcVersion.LessThan(minFMCVersion{{if .MinimumVersionCreate}}Create{{end}}{{camelCase .Name}}) {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("UnsupportedVersion: FMC version %s does not support {{.Name}} creation, minumum required version is {{if .MinimumVersionCreate}}{{.MinimumVersionCreate}}{{else}}{{.MinimumVersion}}{{end}}", r.client.FMCVersion))
```
Comment (danischm) [inline gen/templates/resource.go:455]: same typo

**Review by danischm — CHANGES_REQUESTED** [review]

File: gen/templates/data_source.go (hunk 1 above)
Comment (rchrabas) [reply gen/templates/data_source.go]: fixed

File: gen/templates/resource.go (hunk 2 above)
Comment (rchrabas) [reply gen/templates/resource.go:455]: fixed

**Review by danischm — APPROVED** [review]

**Pull request merged**